File: src-tauri/src/key.rs
use crate::core;

#[tauri::command]
pub fn list_keys() -> Result<Vec<core::models::KeyEntry>, String> {
    core::key::list_keys()
}

#[tauri::command]
pub fn import_key(
    source_path: String,
    name: Option<String>,
    password: Option<String>,
) -> Result<core::models::KeyEntry, String> {
    log::info!(
        "[key] import from file: path={} name={:?} has_password={}",
        source_path,
        name,
        password.is_some()
    );
    core::key::import_key(source_path, name, password)
}

#[tauri::command]
pub fn import_key_content(
    content: String,
    name: String,
    password: Option<String>,
) -> Result<core::models::KeyEntry, String> {
    log::info!(
        "[key] import from content: name={} len={} has_password={}",
        name,
        content.len(),
        password.is_some()
    );
    core::key::import_key_content(content, name, password)
}

#[tauri::command]
pub fn delete_key(id: String) -> Result<(), String> {
    log::info!("[key] delete key: id={}", id);
    core::key::delete_key(id)
}

File: src/types/key.ts
/** 密钥相关类型定义 */

/** 密钥条目 */
export interface KeyEntry {
  id: string;
  name: string;
  key_type: string;
  content: string;
  password?: string | null;
}

/** 通过文件路径导入密钥的参数 */
export interface ImportKeyParams {
  sourcePath: string;
  name: string | null;
  password: string | null;
}

/** 通过内容导入密钥的参数 */
export interface ImportKeyContentParams {
  content: string;
  name: string;
  password: string | null;
}
